### ISSUE TITLE
git-flow

### ISSUE THREAD

**Issue by sielay** [serverless/serverless#310]
Hi

Sorry for being bit critical, but I think honest opinion can be more than constructive. I love your project, your effort and the idea. Problem is process. I tried to install JAWS from npm and both `jaws` and `jaws-framework` seemed to lack `module install` option (as it was removed) and `postinstall` haven't executed at all. I tried to checkout `v1.4` head and I got:

```
TypeError: BbPromise.catch is not a function
    at Object.exports.execute (/usr/local/lib/node_modules/jaws-framework/lib/utils/index.js:61:13)
    at Object.<anonymous> (/usr/local/lib/node_modules/jaws-framework/bin/jaws:11:11)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
    at startup (node.js:117:18)
    at node.js:951:3
```

and when I checked out 1 checking before removing `module install` from master `project create` didn't work.

As you can imagine I was almost going to stop using JAWS. But as we are all early adopters I have bit more will to stay and don't want to reinvent the wheel.

Scary thing is that some people commit directly to master or version branch and seems that there is no review or tests. There are lot of commits like `in progress`. All of it is bit against idea how git should be used.

IMHO (I am not only one) once you decide to do change you should create branch in your fork. Do change, test, change, test, change test. Then once you are happy and all is tested you do `git rebase` to squash all `crappy name commits` and create PR to man repository.

It is a bit overhead, but you would need to adapt it to build sustainable development is such big team. I may be wrong, but current process can only kill the project.

**Comment by HintikkaKimmo** [comment]
Agreed 100%

**Comment by eahefnawy** [comment]
I completely agree with you. Our codebase and workflow is really messy right now. It's mostly because we're still figuring out lots things, specifically for the team & workflow.

We try to keep `master` as clean as possible, and we don't encourage checking out `v1.4` at all, since it'll always be breaking. Once it's complete and tested we'll merge into `master`. This workflow is far from perfect, but we're improving rapidly (got lots of awesome things coming up :wink:)

Thanks so much for your feedback. And please, keep it coming!

**Comment by sielay** [comment]
will do :) project is great
